src/main/java/com/maddyhome/idea/vim/extension/searchindex/VimSearchIndexExtension.kt

```diff
@@ -0,0 +1,183 @@
+  override fun canBeEnabledOn(statusBar: StatusBar) = true
+  override fun isAvailable(project: Project) = true
+}
+
+internal class SearchIndexWidget(project: Project) :
+  EditorBasedWidget(project),
+  StatusBarWidget.TextPresentation {
+
+  override fun ID() = SearchIndex.ID
+  override fun getPresentation() = this
+  override fun getClickConsumer(): Consumer<MouseEvent>? = null
+  override fun getText() = SearchIndex.getCurrentText()
+  override fun getTooltipText() = null
+  override fun getAlignment() = Component.RIGHT_ALIGNMENT
+}
+
+private class SearchAgainHandler(private val keys: List<KeyStroke>) : ExtensionHandler {
+  override fun execute(editor: VimEditor, context: ExecutionContext, operatorArguments: OperatorArguments) {
+    VimExtensionFacade.executeNormalWithoutMapping(keys, editor.ij)
+    VimSearchIndexExtension.updateSearchIndex(editor.ij, VimPlugin.getSearch())
+  }
+}
+
+internal class VimSearchIndexExtension : VimExtension, ModeChangeListener {
+  init {
+    LOG.info("Looking at SearchGroup implementation: ${VimPlugin.getSearch().javaClass}")
+  }
+
+  companion object {
+    private val LOG = logger<VimSearchIndexExtension>()
+
+    fun updateSearchIndex(editor: Editor, searchGroup: IjVimSearchGroup) {
+      if (searchGroup.lastSearchPattern.isNullOrEmpty()) {
+        return
+      }
+      val pattern = searchGroup.lastSearchPattern
+      if (pattern.isNullOrEmpty()) {
+        return
+      }
+
+      LOG.info("search index - Processing pattern: $pattern")
+
+      // FIXME: global search option -> previous search option
+      val ignoreCase = injector.options(editor = editor.vim).ignorecase
+      LOG.info("Ignore case: $ignoreCase")
+
+      val results = injector.searchHelper.findAll(
+        editor = editor.vim,
+        pattern = pattern,
+        startLine = 0,
+        endLine = -1,
+        ignoreCase = ignoreCase,
+      )
+
+      val label = ExEntryPanel.getInstance().label
+      val (currentIndex: Int, resultSize: Int) = if (results.isNotEmpty()) {
+        Pair(first = getCurrentIndex(editor = editor, results = results), second = results.size)
+      } else {
+        Pair(first = 0, second = 0)
+      }
+
+      ApplicationManager.getApplication().invokeLater {
+        LOG.info("searchindex - [${currentIndex}/${resultSize}] $label$pattern")
+        SearchIndex.update("[${currentIndex}/${resultSize}] $label$pattern")
+      }
+    }
+
+    private fun getCurrentIndex(
+      editor: Editor,
+      results: List<TextRange>,
+    ): Int {
+      val caretOffset = editor.caretModel.offset
+      var current = 0
+      results.forEachIndexed { index, range ->
+        if (range.startOffset <= caretOffset && caretOffset <= range.endOffset) {
+          current = index + 1
+        }
+      }
+      return current
+    }
+  }
+
+  override fun getName(): String = "searchindex"
+
+  override fun init() {
+    LOG.info("Initializing SearchIndex extension")
```

Comment:
The INFO level of logging is too verbose and will clutter the log file. Let’s change it to “DEBUG.”

But I appreciate you considering it!